lib/cublas/wrappers.jl

```diff
@@ -126,17 +126,29 @@ for (fname, fname_64, elty) in ((:cublasDscal_v2, :cublasDscal_v2_64, :Float64),
         end
     end
 end
+function scal!(n::Integer, alpha::Number, x::StridedCuVecOrDenseMat{T}) where {T}
+    α = convert(T, alpha)
+    gpu_α = CuRef{T}(α)
+    scal!(n, gpu_α, x)
+    synchronize()
```

Comment:
Why the synchronization? The only way to see the changes by this call is to fetch memory, which is a synchronizing operation.

Reply:
For `scal!` and other functions which don't return a scalar result, I added this to keep the previous behaviour (so that the entire call is synchronous). I'll remove the sync for things like `nrm2!` that return a scalar which is copied back anyway.

Reply:
> For `scal!` and other functions which don't return a scalar result, I added this to keep the previous behaviour (so that the entire call is synchronous).

That kind of defeats the purpose of this change, no? Since this is operating on GPU memory, which can't be fetched without synchronizing anyway, I don't see why we should keep the previous synchronous behavior. Making it asynchronous shouldn't break anything.

Reply:
In fact, this method should have been asynchronous before (as it doesn't return a scalar, https://docs.nvidia.com/cuda/cublas/#scalar-parameters), so synchronizing now seems like a regression.